File: Week_03/111/minimum_depth_of_binary_tree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 1
        elif root.left is not None and root.right is not None:
            return 1 + min(self.minDepth(root.left), self.minDepth(root.right))
        elif root.left is not None:
            return 1 + self.minDepth(root.left)
        else:
            return 1 + self.minDepth(root.right)

File: Week_03/111/minimum_depth_of_binary_tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func minDepth(node *TreeNode) int {
	if node == nil {
		return 0
	}
	if node.Left == nil && node.Right == nil{
		return 1
	} else if node.Left != nil && node.Right != nil {
		return 1 + min(minDepth(node.Left), minDepth(node.Right))
	} else if node.Left != nil {
		return 1 + minDepth(node.Left)
	} else {
		return 1 + minDepth(node.Right)
	}
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}